### PR TITLE
[WIP] CSS Framework Typography & Breakpoint Scale Rework

### PR DESCRIPTION
**Pull request by hxltrhuxze**
NOTE: This PR is larger than others. This is unfortunately because the breakpoint and typography scale are heavily interlinked, so both had to change simultaneously.

## Overview

This PR changes the Typography Scale and Breakpoint Scale to maintain visual consistency with the Sketch toolkit.

## Reason

The Typography scale's `base` value was set to `16px`, which is too large for the Sketch Toolkit. This PR moves each scale position down one, giving two benefits:
1. Makes the `base` scale position `0.875rem` (`14px` at body `font-size: 100%`). 
2. Balances the scale so that it runs `xxs` through `xxl`, rather than `xxxs` through `xl`.

| Updated Scale | Current Scale | Computed Value (rems) |
|----------------|---------------|------------------|
| `xxs`                      | `xxxs`                 | `0.5`                      |
| `xs`                        | `xxs`                   | `0.625`                 |
| `s`                          | `xs`                     | `0.75`                   |
| `base`                   | `s`                       | `0.875`                 |
| `m`                        | `base`                 | `1`                         |
| `l`                          | `m`                       | `1.125`                  |
| `xl`                        | `l`                         | `1.25`                    |
| `xxl`                      | `xl`                       | `1.5`                      |
| `display`               | `display`            | `1.875`                 |
| `display-l`             | `display-l`          | `2.25`                  |
| `display-xl`           | `display-xl`        | `3`                       |

---

The Breakpoint Scale adjusts the `font-size` of the entire web page at different screen sizes. The original scale had the base `font-size: 100%` set on the smallest breakpoint `0 - 575px`. This scaled upwards at wider breakpoints, causing the larger screen-sizes to diverge from the Sketch Toolkit. This PR normalises this scale, placing `font-size: 100%` at the `1200px - 1400px` breakpoint.

| Breakpoint Scale  | Previous Font Size (%) | Updated Font Size (%) |
|----------------|---------------|------------------|
| `root`                 | `100`               | `92`                      |
| `xs`                    | `-`                    | `94`                      |
| `s`                      | `103`               | `96`                      |
| `m`                     | `106`               | `98`                     |
| `l`                       | `106`               | `100`                    |
| `xl`                     | `112`                | `103`                    |
| `xxl`                   | `115`                | `106`                    |
| `xxxl`                 | `118`                | `-`                         |

## Work carried out

- [x] Adjusted the Typography scale positions.
- [x] Changed components using the `font-size` function to the new scale position:
    - [x] Avatar
    - [x] Badge
    - [x] Breadcrumbs
    - [x] Button
    - [x] Checkbox
    - [x] Dialog
    - [x] Dropdown
    - [x] Number Input
    - [x] Radio
    - [x] Select
    - [x] Text Input
    - [x] Textarea
    - [x] Masthead
    - [x] Modal
    - [x] Nav
    - [x] Pagination
    - [x] Phase Banner
    - [x] Sidebar
    - [x] Switch
    - [x] Tab Set
    - [x] Table
    - [x] Tooltip
- [x] Changed all instances of breakpoint utility classes.
- [x] Changed all instances of typography utility classes.
- [x] Adjusted the Breakpoint Scale positions & values.
- [x] Updated Typography Docs to match new scale.
- [x] Updated Breakpoint Docs to match new scale.

### CODE REVIEW

File: packages/docs-site/src/components/presenters/swatch/Swatch.scss [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 .swatch-label {
   text-align: center;
-  font-size: 12px;
+  font-size: font-size(s);
```
Comment (m7kvqbe1) [inline packages/docs-site/src/components/presenters/swatch/Swatch.scss]: This should be in quotes?

**Review by m7kvqbe1 — APPROVED** [review]
Just one thing

**Comment by m7kvqbe1** [comment]
Shall we squash all those commits on merge.

File: packages/docs-site/src/components/presenters/swatch/Swatch.scss (hunk 1 above)
Comment (hxltrhuxze) [reply packages/docs-site/src/components/presenters/swatch/Swatch.scss]: good spot, thanks

**Comment by hxltrhuxze** [comment]
yeah 100%, we'll squash

**Pull request merged**